database/postgres/postgres_collector.go

```diff
@@ -0,0 +1,294 @@
+package postgres
+
+import (
+	"github.com/frain-dev/convoy/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"strings"
+	"time"
+)
+
+// Namespace used in fully-qualified metrics names.
+const namespace = "convoy"
+
+const delaySeconds = 1
+
+var lastRun = time.Now()
+
+type EventQueueMetrics struct {
+	ProjectID string `json:"project_id" db:"project_id"`
+	SourceId  string `json:"source_id" db:"source_id"`
+	Total     uint64 `json:"total" db:"total"`
+}
+
+type EventQueueBacklogMetrics struct {
+	ProjectID  string  `json:"project_id" db:"project_id"`
+	SourceId   string  `json:"source_id" db:"source_id"`
+	AgeSeconds float64 `json:"age_seconds" db:"age_seconds"`
+}
+
+type EventDeliveryQueueMetrics struct {
+	ProjectID  string `json:"project_id" db:"project_id"`
+	EndpointId string `json:"endpoint_id" db:"endpoint_id"`
+	Status     string `json:"status" db:"status"`
+	Total      uint64 `json:"total" db:"total"`
+}
+
+type EventDeliveryQueueLatencyMetrics struct {
+	ProjectID      string `json:"project_id" db:"project_id"`
+	EndpointId     string `json:"endpoint_id" db:"endpoint_id"`
+	Status         string `json:"status" db:"status"`
+	LatencySeconds uint64 `json:"latency_seconds" db:"latency_seconds"`
+}
+
+type EventQueueEndpointBacklogMetrics struct {
+	ProjectID  string  `json:"project_id" db:"project_id"`
+	SourceId   string  `json:"source_id" db:"source_id"`
+	EndpointId string  `json:"endpoint_id" db:"endpoint_id"`
+	AgeSeconds float64 `json:"age_seconds" db:"age_seconds"`
+}
+
+type EventQueueEndpointAttemptMetrics struct {
+	ProjectID  string `json:"project_id" db:"project_id"`
+	EndpointId string `json:"endpoint_id" db:"endpoint_id"`
+	Status     string `json:"status" db:"status"`
+	StatusCode string `json:"status_code" db:"status_code"`
+	Total      uint64 `json:"total" db:"total"`
+}
+
+type Metrics struct {
+	EventQueueMetrics []EventQueueMetrics
+
+	//
+	EventQueueBacklogMetrics         []EventQueueBacklogMetrics
+	EventDeliveryQueueMetrics        []EventDeliveryQueueMetrics
+	EventQueueEndpointBacklogMetrics []EventQueueEndpointBacklogMetrics
+	EventQueueEndpointAttemptMetrics []EventQueueEndpointAttemptMetrics
+}
+
+var (
+	eventQueueTotalDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "event_queue_total"),
+		"Total number of tasks in the event queue",
+		[]string{"project_id", "source", "status"}, nil,
+	)
+
+	eventQueueBacklogDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "event_queue_backlog_seconds"),
+		"Number of seconds the oldest pending task is waiting in pending state to be processed.",
+		[]string{"project_id", "source"}, nil,
+	)
+
+	eventDeliveryQueueTotalDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "convoy_event_delivery_queue_total"),
```

Comment:
@mekilis, this shouldn't have the prefix `convoy` since it's already in the namespace.

Reply:
Done